src/Utils/ObjectMixin.php

```diff
@@ -317,6 +306,45 @@ public static function getMagicMethods($class)
 
 
 	/**
+	 * Returns array of magic properties defined by annotation @property.
+	 * @return array
+	 */
+	public static function getMagicProperties($class)
+	{
+		$props = & self::$properties[$class];
+		if ($props !== NULL) {
+			return $props;
+		}
+
+		$rc = new \ReflectionClass($class);
+		preg_match_all('~^
+			[ \t*]*  @property(|-read|-write)  [ \t]+
+			(?: [^\s]+  [ \t]+ )?
+			\$ (\w+)
+		()~mx', (string) $rc->getDocComment(), $matches, PREG_SET_ORDER);
+
+		$props = [];
+		foreach ($matches as list(, $type, $name)) {
+			$uname = ucfirst($name);
+			$write = (!$type || $type === '-write')
+				&& $rc->hasMethod($nm = 'set' . $uname)
+				&& ($rm = $rc->getMethod($nm)) && $rm->getName() === $nm && !$rm->isPrivate() && !$rm->isStatic();
+			$read = (!$type || $type === '-read')
+				&& ($rc->hasMethod($nm = 'get' . $uname) || $rc->hasMethod($nm = 'is' . $uname))
+				&& ($rm = $rc->getMethod($nm)) && $rm->getName() === $nm && !$rm->isPrivate() && !$rm->isStatic();
+			if ($read || $write) {
+				$props[$name] = [$read ? $nm : FALSE, $write, $read ? $rm->returnsReference() : NULL];
```

Comment:
Good point, that would require another bit flag. The `__get` initial calling sequence would then look as

~~~php
$class = get_class($this);
$uname = ucfirst($name);
$prop = ObjectMixin::getMagicProperty($class, $name);

if ($name === '') {
    throw new MemberAccessException("Cannot read a class '$class' property without name.");

} elseif ($prop & BIT_MAGIC_A) {
    return $this->{'get' . $uname}();

} elseif ($prop & BIT_MAGIC_B) {
    return $this->{'is' . $uname}();

} elseif ($prop & BIT_MAGIC_C) {
    $value = $this->{'get' . $uname}();
    return $value;

} elseif ($prop & BIT_MAGIC_D) {
    $value = $this->{'is' . $uname}();
    return $value;
}
~~~

Yeah, I know, over-optimization but that's just how I like it =)